src/dataFlattener.ts

```diff
@@ -0,0 +1,66 @@
+// Needs typing (I originally wrote this in javascript)
+
+export const dataFlattener = (nestedData) => {
+  type flatData = {
+    code: string;
+    name: string;
+    type: string;
+    country: string;
+    subnational1: string | null;
+    subnational2: string | null;
+    fullHierarchyName: string;
+  };
+
+  type region = {
+    code: string;
+    name: string;
+    subnational1: string | undefined;
+    subnational2: string | undefined;
+  };
+
+  const flattenedData: flatData[] = [];
+
+  function recurse(
```

Comment:
nit naming: calling this recurse is a little generic, it would better to describe exactly what it does with the name, if possible.

For instance in this case we could name it something like `flattenNestedRegions`, which tells us what will happen with this function when we pass it a region.

That said, I reckon there's a way to write this script that doesn't need to be recursive, which might make it simpler to maintain going forward.

Reply:
I've renamed the recurse function to flattenNestedRegions. We'll get on a call and address the remaining points!